app/__init__.py

```diff
@@ -45,15 +60,40 @@ def create_app(config_class=Config):
     health = Healthcheck(app)
     health.add_check(FlaskRunningChecker())
 
-    # TODO: TOWNS_FUND_AUTH is currently stored in const.py but this isn't isn't a good solution.
-    #   We need to decide where we should store and inject specific auth mappings from.
-    app.logger.info("Setting up auth")
-    email_mapping = copy(TOWNS_FUND_AUTH)
-    app.logger.info(f"Additional auth details from secret: {Config.ADDITIONAL_EMAIL_LOOKUPS}")
-    email_mapping.update(Config.ADDITIONAL_EMAIL_LOOKUPS)
-    app.config["AUTH_MAPPING"]: AuthMapping = build_auth_mapping(Config.FUND_NAME, email_mapping)
+    setup_funds_and_auth(app)
 
     return app
 
 
+def setup_funds_and_auth(app: Flask) -> None:
+    """Sets up the funding config and auth mappings.
+
+    - FUND_CONFIGS: maps user roles to a fund configuration
+        - this injects fund specific context into the rest of the application based on the current users role
+    - AUTH_MAPPINGS: maps funds to sets of authorisation details that define what users can submit
+        - these authorisation details are passed to the backend to check against the submitted return
+
+    TODO: Going forwards the logic and state for "auth" and "fund" config should be extracted from this repo and
+      encapsulated in separate microservices with their own databases.
+      This current mono-repo implementation with state stored in code (see _TF_FUND_CONFIG and _TOWNS_FUND_AUTH) works
+      for now but should not be seen as a long term solution.
+
+    :param app: the Flask app
+    :return: None
+    """
+    app.logger.info("Setting up fund configs and auth mappings")
+    app.logger.info(f"Additional auth details from secret: {str(Config.ADDITIONAL_EMAIL_LOOKUPS)}")
+
+    # funds
+    towns_fund: FundConfig = copy(TOWNS_FUND_APP_CONFIG)
+    app.config["FUND_CONFIGS"] = ReadOnlyFundConfigs(role_to_fund_configs={towns_fund.user_role: towns_fund})
+
+    # auth
+    tf_auth = copy(TOWNS_FUND_AUTH)
```

Comment:
Could do wit a comment explaining how these are constructed here. copy is okay over deepcopy?

Reply:
> Could do wit a comment explaining how these are constructed here.

How do you mean constructed? 

Good spot on the copy vs deepcopy

Reply:
Sorry, not very clear. I guess taking a copy implies that you are going to mutate either the inital TF_APP_CONFIG or the newly created towns_fund and you need to keep track of both versions. But I can't see that happening. why not just pass

```towns_fund = TOWNS_FUND_APP_CONFIG```

I see that you update the dict in the TF_AUTH case so that makes sense. I think deepcopy and copy are effectively the same here anyway since TF_AUTH is just a dict of tuple

Reply:
Yeah upon reflection I agree with your points - this is just going too far to protect against modification of the constants